src/chains/ethereum/ethereum/src/blockchain.ts

```diff
@@ -577,6 +583,40 @@ export default class Blockchain extends Emittery<BlockchainTypedEvents> {
 
   coinbase: Address;
 
+  gasEstimateBlock = async (
```

Comment:
it'd be nice if this were a verb, ie `createEstimateGasBlock` or something. 

I haven't the time to look into this further today, but I wonder if there's some value in combining the `readyNextBlock` and this function, as they are very similar (it might be that it just becomes too complicated to be worthwhile though).